File: update_tracker.py
import re

with open('docs/game_architecture_plan.md', 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
in_target_section = False

target_pattern = re.compile(r'### 🟠 Listening|### 🟡 Speaking|### 🟤 Writing|### 🔴 Accent|### 🎭 Roleplay|### 👑 Elite Mastery')
game_title_pattern = re.compile(r'^\d+\.\s+\*\*(.*)\*\*$')

diamond_checklist = [
    "- [ ] Dual-Stage Scroll UX\n",
    "- [ ] Sliver Performance Layout\n",
    "- [ ] Zero setState (ValueNotifier)\n",
    "- [ ] Feedback Card Logic\n",
    "- [ ] Pedagogical Component: `TBD`\n",
    "- [ ] Edge-to-Edge Scrollbar (RawScrollbar)\n",
    "- [ ] Keyboard Scroll Stability (FocusNode visibility)\n",
    "- [ ] Docked Input Padding (Keyboard Games Only)\n",
    "- [ ] 10/10 UX Confirmation (No Patchwork)\n",
    "- [ ] Git Commit & Push\n"
]

skip_lines = False

for line in lines:
    if target_pattern.search(line):
        in_target_section = True
        
    if in_target_section:
        if line.startswith("- [x] ") or line.startswith("- [ ] "):
            # Skip existing checkboxes
            continue
        
        match = game_title_pattern.search(line.strip())
        if match:
            new_lines.append(line)
            new_lines.extend(diamond_checklist)
            continue
            
    new_lines.append(line)

with open('docs/game_architecture_plan.md', 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

print("Updated game_architecture_plan.md")

File: update_tracker.dart
import 'dart:io';

void main() async {
  final file = File('docs/game_architecture_plan.md');
  final lines = await file.readAsLines();

  final newLines = <String>[];
  bool inTargetSection = false;

  final targetPattern = RegExp(r'### 🟠 Listening|### 🟡 Speaking|### 🟤 Writing|### 🔴 Accent|### 🎭 Roleplay|### 👑 Elite Mastery');
  final gameTitlePattern = RegExp(r'^\d+\.\s+\*\*(.*)\*\*$');

  final diamondChecklist = [
    "- [ ] Dual-Stage Scroll UX",
    "- [ ] Sliver Performance Layout",
    "- [ ] Zero setState (ValueNotifier)",
    "- [ ] Feedback Card Logic",
    "- [ ] Pedagogical Component: `TBD`",
    "- [ ] Edge-to-Edge Scrollbar (RawScrollbar)",
    "- [ ] Keyboard Scroll Stability (FocusNode visibility)",
    "- [ ] Docked Input Padding (Keyboard Games Only)",
    "- [ ] 10/10 UX Confirmation (No Patchwork)",
    "- [ ] Git Commit & Push"
  ];

  for (final line in lines) {
    if (targetPattern.hasMatch(line)) {
      inTargetSection = true;
    }

    if (inTargetSection) {
      if (line.startsWith("- [x] ") || line.startsWith("- [ ] ")) {
        continue;
      }

      final match = gameTitlePattern.firstMatch(line.trim());
      if (match != null) {
        newLines.add(line);
        newLines.addAll(diamondChecklist);
        continue;
      }
    }

    newLines.add(line);
  }

  await file.writeAsString(newLines.join('\n'));
  print('Updated game_architecture_plan.md');
}
